### PR TITLE
JIT: ARM64 SVE format encodings, `SVE_BX_2A` and `SVE_BY_2A`

### PR DESCRIPTION
**Pull request by TIHan**
Contributes to https://github.com/dotnet/runtime/issues/94549

Adds formats:
- SVE_BX_2A
- SVE_BY_2A

Left: Capstone
Right: Jit
image

### CODE REVIEW

**Comment by TIHan** [comment]
@dotnet/jit-contrib @dotnet/arm64-contrib @a74nh @amanasifkhalid this is ready.

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks!

**Review by a74nh — APPROVED** [review]
LGTM

File: src/coreclr/jit/emitarm64.cpp [hunk 1]
```diff
@@ -2626,6 +2626,48 @@ void emitter::emitInsSanityCheck(instrDesc* id)
...
             break;
 
+        case IF_SVE_BX_2A: // ...........ixxxx ......nnnnnddddd -- sve_int_perm_dupq_i
+            imm      = emitGetInsSC(id);
+            elemsize = id->idOpSize();
+            assert(insOptsScalableStandard(id->idInsOpt()));
+            assert(isVectorRegister(id->idReg1()));
+            assert(isVectorRegister(id->idReg2()));
+            assert(isScalableVectorSize(elemsize));
+#ifdef DEBUG
```
Comment (kunalspathak) [inline src/coreclr/jit/emitarm64.cpp:2636]: don't think you need `#ifdef DEBUG` here. All these has is `assert` and compiler should be able to eliminate it.

**Review by kunalspathak — APPROVED** [review]
LGTM

File: src/coreclr/jit/emitarm64.cpp (hunk 1 above)
Comment (TIHan) [reply src/coreclr/jit/emitarm64.cpp:2636]: Yea, this has been commented about before. Personally, I like to have the `ifdef` for these situations as it distinguishes code that is for verification. Asserts on their own obviously don't need an `ifdef`, but for checks that require branching, I tend to wrap them around an `ifdef`.

**Pull request merged**